python/src/iceberg/expressions.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from enum import Enum
+
+from iceberg.exceptions import OperationError
+
+
+class Operation(Enum):
+    """Operations to be used as components in expressions
+
+    Various operations can be negated or reversed. Negating an
+    operation is as simple as using the built-in subtraction operator:
+
+    >>> print(-Operation.TRUE)
+    Operation.FALSE
+    >>> print(-Operation.IS_NULL)
+    Operation.NOT_NULL
+
+    Reversing an operation can be done using the built-in reversed() method:
+    >>> print(reversed(Operation.LT))
+    Operation.GT
+    >>> print(reversed(Operation.EQ))
+    Operation.NOT_EQ
+
+    Raises:
+        OperationError: This is raised when attempting to negate or reverse
+            an operation that cannot be negated or reversed.
+    """
+    TRUE = "TRUE"
+    FALSE = "FALSE"
+    IS_NULL = "IS_NULL"
+    NOT_NULL = "NOT_NULL"
+    IS_NAN = "IS_NAN"
+    NOT_NAN = "NOT_NAN"
+    LT = "LT"
+    LT_EQ = "LT_EQ"
+    GT = "GT"
+    GT_EQ = "GT_EQ"
+    EQ = "EQ"
+    NOT_EQ = "NOT_EQ"
+    IN = "IN"
+    NOT_IN = "NOT_IN"
+    NOT = "NOT"
+    AND = "AND"
+    OR = "OR"
+
+    def __str__(self):
+        return self.value
+
+    def __repr__(self):
+        return f"Operation.{self.value}"
```

Comment:
According to this doc, you should get __repr__ and __str__ for free: https://docs.python.org/3.11/howto/enum.html

Reply:
Nice! I never noticed that. I'll update this.

Reply:
removed